### PR TITLE
chore: update docs for Evaluate Async

### PR DESCRIPTION
**Pull request by JimLiu**

### CODE REVIEW

File: api/wd/index.md [hunk 1]
```diff
@@ -525,6 +525,38 @@ browser.evaluate(function fakeConfigData() {
...
 });
+
+#### `browser.evaluateAsync(args..., function(args...))`
+
+The executed script is assumed to be asynchronous.
+
+Returns a Promise of the given function, invoked on the webdriver side (so you can not bind its `this` object or access context variables via lexical closure).
+If you provided `args`, testium will marshals them as JSON and passes them to the function in the given order.
+
+Note: NOT support Phantomjs
```
Comment (dbushong) [inline api/wd/index.md]:
```suggestion
Note: NOT supported on PhantomJS
```

File: api/wd/index.md [hunk 2]
```diff
@@ -525,6 +525,38 @@ browser.evaluate(function fakeConfigData() {
 browser.evaluate('hash', function getLocationProp(prop) {
   return window.location[prop];
 });
+
+#### `browser.evaluateAsync(args..., function(args...))`
+
+The executed script is assumed to be asynchronous.
+
+Returns a Promise of the given function, invoked on the webdriver side (so you can not bind its `this` object or access context variables via lexical closure).
```
Comment (dbushong) [inline api/wd/index.md]: We try to keep the `.md` files wrapped around 80 cols:
```suggestion
Returns a Promise of the given function, invoked on the webdriver side (so you
can not bind its `this` object or access context variables via lexical closure).
```

File: api/wd/index.md [hunk 3]
```diff
@@ -525,6 +525,38 @@ browser.evaluate(function fakeConfigData() {
...
+If you provided `args`, testium will marshals them as JSON and passes them to the function in the given order.
```
Comment (dbushong) [inline api/wd/index.md]:
```suggestion
If you provided `args`, testium will marshals them as JSON and passes them to
the function in the given order.
```

File: api/wd/index.md [hunk 4]
```diff
@@ -525,6 +525,38 @@ browser.evaluate(function fakeConfigData() {
 browser.evaluate('hash', function getLocationProp(prop) {
   return window.location[prop];
 });
+
+#### `browser.evaluateAsync(args..., function(args...))`
+
+The executed script is assumed to be asynchronous.
```
Comment (dbushong) [inline api/wd/index.md]:
```suggestion
The executed script is assumed to return a Promise.
```

File: api/wd/index.md [hunk 5]
```diff
@@ -525,6 +525,38 @@ browser.evaluate(function fakeConfigData() {
...
+Note: NOT support Phantomjs
+
+##### Example: Fetch the data
+
+```js
+// Fetching data from the client
+browser.evaluate(() => {
+  return fetch('http://example.com/movies.json')
+    .then(response => response.json());
+});
```
Comment (dbushong) [inline api/wd/index.md]: nice example - maybe skip the `{ return }` and just use implict return to make it super-pretty?

**Comment by dbushong** [comment]
nice; few comments & suggestions.

might be nice (after verifying it works) to add an example where you pass an `async () => { }` - it *should* work

File: api/wd/index.md (hunk 5 above)
Comment (JimLiu) [reply api/wd/index.md]: @dbushong 
I keep the `return` just want to remind they don't forget to return a `Promise` :)

**Comment by JimLiu** [comment]
Updated

File: api/wd/index.md (hunk 5 above)
Comment (JimLiu) [reply api/wd/index.md]: Add the `return` in the async example

File: api/wd/index.md [hunk 6]
```diff
@@ -525,6 +525,49 @@ browser.evaluate(function fakeConfigData() {
 browser.evaluate('hash', function getLocationProp(prop) {
   return window.location[prop];
 });
+
+#### `browser.evaluateAsync(args..., function(args...))`
```
Comment (dbushong) [inline api/wd/index.md]: this is inside the ```js block above, and will render wrong

Comment (JimLiu) [reply api/wd/index.md]: Good catch, updated :P

**Comment by dbushong** [comment]
Thanks; great addition!

**Pull request merged**